src/storage/data_table.cpp

```diff
@@ -137,23 +155,20 @@ bool DataTable::Update(const common::ManagedPointer<transaction::TransactionCont
   return true;
 }
 
-void DataTable::CheckMoveHead(std::list<RawBlock *>::iterator block) {
-  // Assume block is full
+void DataTable::CheckMoveHead(uint32_t block_index) {
+  // Assume block is full.
   common::SpinLatch::ScopedSpinLatch guard_head(&header_latch_);
```

Comment:
It seems like this is an unnecessary latch since the insertion head is now an atomic integer rather than a list iterator.